use /usr/workspace/benchpark for gitlab data files

slabasan opened the pull request:
The gitlab CI has filled up the `benchpark` user's disk quota. See example https://lc.llnl.gov/gitlab/benchpark/benchpark/-/jobs/2487244
```
=> Warning: Errors occurred when archiving files.
	See: /g/g0/benchpark/.jacamar-ci/builds/jyvWCxYV/001/gitlab/benchpark/benchpark/workspace/spack/opt/spack/linux-rhel8-zen3/cce-18.0.1-rocm6.2.4/chai-2024.02.2-nzn47myhb5jp5vplnoeboyfw6ma5tlw2/.spack/archived-files/errors.txt
==> Error: OSError: [Errno 122] Disk quota exceeded
```

This change would put the CI files in `/usr/workspace/benchpark`.

slabasan committed:
use /usr/workspace/benchpark for gitlab data files

michaelmckinsey1 reviewed (commented):
@slabasan I wonder if it would be better to `rm -rf /g/g0/benchpark/.jacamar-ci/builds` after the `report_status` step (aka all 9 jobs are done). Are there a bunch of builds in `/g/g0/benchpark/.jacamar-ci/builds`? I think even changing the directory will fill up that directory as well, since we are cloning spack every time.

slabasan committed:
more fixes

- remove extra newline
- remove quotation marks
- make spack install path less deep

pearce8 reviewed (changes requested):
@slabasan @michaelmckinsey1 Are we ready to merge this?

pearce8 committed:
Merge branch 'develop' into gitlab-update-working-dir

michaelmckinsey1 commented:
> @slabasan @michaelmckinsey1 Are we ready to merge this?

I don't think this is working currently

```
==> Error: SbangPathError: Install tree root is too long. Spack cannot patch shebang lines when script path length (128) exceeds limit (127).
```

We did add a symlink a couple weeks ago from home to `/usr/workspace/benchpark`, which has accomplished the goal of this PR, so maybe we close this one @slabasan ?
